monitor/src/main.rs

```diff
@@ -11,11 +11,11 @@
 #![cfg_attr(not(test), no_main)]
 #![cfg_attr(not(test), no_std)]
 
+libhypatia::define_segment!(init);
+
+// XXX(mikew): For some reason, removing this no_mangle on this init in particular causes
```

Comment:
Hmm. I've got to poke at this a bit more; off the top of my head, I'm not sure what's going on; maybe I can ask Steve Klabnik tomorrow. I don't think we should block on this, though (I'm also going out of town for a week in a couple of days). It certainly is worth running to ground: I confess a lot of this sort of stuff leaves me somewhat mystified and I'd like to understand more.